src/caja-location-bar.c

```diff
@@ -263,8 +267,17 @@ style_set_handler (GtkWidget *widget, GtkStyle *previous_style)
     pango_layout_get_pixel_size (layout, &width2, NULL);
     width = MAX (width, width2);
 
+#if GTK_CHECK_VERSION (3, 0, 0)
+    gtk_widget_get_margin_start (widget);
+    gtk_widget_get_margin_end (widget);
+    margin_top = gtk_widget_get_margin_top (widget);
+    margin_bottom = gtk_widget_get_margin_bottom (widget);
+    xpad = margin_start + margin_end;
+    ypad = margin_top + margin_bottom;
```

Comment:
just taken from marco fix for it.
Please correct all this
... i'm tired of of working on caja after 4 weeks